stdlib/builtin/hash.rbs

```diff
@@ -1,410 +1,1033 @@
+# and the two objects are `eql?` to each other.
+#
 # A user-defined class may be used as a hash key if the `hash` and `eql?`
-# methods are overridden to provide meaningful behavior. By default,
-# separate instances refer to separate hash keys.
-# 
-# A typical implementation of `hash` is based on the object’s data while
-# `eql?` is usually aliased to the overridden `==` method:
-# 
-# ```ruby
-# class Book
-#   attr_reader :author, :title
-# 
-#   def initialize(author, title)
-#     @author = author
-#     @title = title
-#   end
-# 
-#   def ==(other)
-#     self.class === other and
-#       other.author == @author and
-#       other.title == @title
-#   end
-# 
-#   alias eql? ==
-# 
-#   def hash
-#     @author.hash ^ @title.hash # XOR
-#   end
-# end
-# 
-# book1 = Book.new 'matz', 'Ruby in a Nutshell'
-# book2 = Book.new 'matz', 'Ruby in a Nutshell'
-# 
-# reviews = {}
-# 
-# reviews[book1] = 'Great reference!'
-# reviews[book2] = 'Nice and compact!'
-# 
-# reviews.length #=> 1
-# ```
-# 
-# See also Object\#hash and
-# [Object\#eql?](https://ruby-doc.org/core-2.6.3/Object.html#method-i-eql-3F)
+# methods are overridden to provide meaningful behavior.  By default, separate
+# instances refer to separate hash keys.
+#
+# A typical implementation of `hash` is based on the object's data while `eql?`
+# is usually aliased to the overridden `==` method:
+#
+#     class Book
+#       attr_reader :author, :title
+#
+#       def initialize(author, title)
+#         @author = author
+#         @title = title
+#       end
+#
+#       def ==(other)
+#         self.class === other and
+#           other.author == @author and
+#           other.title == @title
+#       end
+#
+#       alias eql? ==
+#
+#       def hash
+#         @author.hash ^ @title.hash # XOR
+#       end
+#     end
+#
+#     book1 = Book.new 'matz', 'Ruby in a Nutshell'
+#     book2 = Book.new 'matz', 'Ruby in a Nutshell'
+#
+#     reviews = {}
+#
+#     reviews[book1] = 'Great reference!'
+#     reviews[book2] = 'Nice and compact!'
+#
+#     reviews.length #=> 1
+#
+# See also Object#hash and Object#eql?
+#
 class Hash[unchecked out K, unchecked out V] < Object
   include Enumerable[[K, V], Hash[K, V]]
 
-  def self.[]: [U, V] (*::Array[[ U, V ]] arg0) -> ::Hash[U, V]
+  # Creates a new hash populated with the given objects.
+  #
+  # Similar to the literal `{ *key* => *value*, ... }`. In the first form, keys
+  # and values occur in pairs, so there must be an even number of arguments.
+  #
+  # The second and third form take a single argument which is either an array of
+  # key-value pairs or an object convertible to a hash.
+  #
+  #     Hash["a", 100, "b", 200]             #=> {"a"=>100, "b"=>200}
+  #     Hash[ [ ["a", 100], ["b", 200] ] ]   #=> {"a"=>100, "b"=>200}
+  #     Hash["a" => 100, "b" => 200]         #=> {"a"=>100, "b"=>200}
+  #
+  def self.[]: [U, V] (_ToHash[U, V]) -> ::Hash[U, V]
+             | [U, V] (*[ U, V ]) -> ::Hash[U, V]
```

Comment:
```
| [U, V] (Array[U, V]) -> Hash[U, V]
```

???

Reply:
Did you mean `[U, V] (Array[[ U, V ]]) -> ::Hash[U, V]`? I fixed with it in 384627caf564dd4051dbde1a3a9fed43a04bf44c.